### PR TITLE
Substitute this with self

### PR DESCRIPTION
**Pull request by hubertp**
### Pull Request Description

A semi-manual s/this/self appied to the whole standard library.
Related to https://www.pivotaltracker.com/story/show/182328601

In the compiler promoted to use constants instead of hardcoded
`this`/`self` whenever possible.

### Important Notes

The PR **does not** require explicit `self` parameter declaration for methods as this part 
of the design is still under consideration.

### Checklist

Please include the following checklist in your PR:

- [ ] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide dist` and `./run ide watch`.

### CODE REVIEW

**Comment by hubertp** [comment]
I dare anyone to review `distribution/lib`. I double dare you ;)

**Comment by radeusgd** [comment]
> I dare anyone to review `distribution/lib`. I double dare you ;)

I've got a long train ride tomorrow evening, so challenge accepted ^^

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Json.enso [hunk 1]
```diff
@@ -183,7 +183,7 @@ type No_Such_Field_Error field_name
    Pretty prints the no such field error.
 No_Such_Field_Error.to_display_text : Text
 No_Such_Field_Error.to_display_text =
-    "The field " + this.field_name.to_text + " is not present in this object."
+    "The field " + self.field_name.to_text + " is not present in self object."
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Json.enso]:
```suggestion
    "The field " + self.field_name.to_text + " is not present in this object."
```

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso [hunk 2]
```diff
@@ -276,13 +276,13 @@ type Number
...
     format : Text -> Text
-    format fmt = String.format fmt this
+    format fmt = String.format fmt self
 
     ## Checks equality of numbers, using an `epsilon` value.
 
        Arguments:
        - that: The number to check equality against.
-       - epsilon: The value by which `this` and `that` can be separated by before
+       - epsilon: The value by which `self` and `that` can be separated by before
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso:285]: I wonder what we should do about these?

`this` and `that` sounded nicely, now, not sure if we should change `that` to something else? But I'm not sure if there are good alternatives either.

File: distribution/lib/Standard/Base/0.0.0-dev/src/System/File/Option.enso [hunk 3]
```diff
@@ -45,9 +45,9 @@ type Option
 
     ## PRIVATE
 
-       Convert this object into a representation understandable by the JVM.
+       Convert self object into a representation understandable by the JVM.
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/System/File/Option.enso]:
```suggestion
       Convert this object into a representation understandable by the JVM.
```

File: distribution/lib/Standard/Database/0.0.0-dev/src/Data/Column.enso [hunk 4]
```diff
@@ -357,11 +357,11 @@ type Column
        - other: The other column to compute the conjunction with.
 
        Returns a column containing the result of performing the boolean `and` on
-       `other` and each element of `this`.  If `other` is a column, the
-       operation is performed pairwise between corresponding elements of `this` ]
+       `other` and each element of `self`.  If `other` is a column, the
+       operation is performed pairwise between corresponding elements of `self` ]
```
Comment (radeusgd) [inline distribution/lib/Standard/Database/0.0.0-dev/src/Data/Column.enso]:
```suggestion
       operation is performed pairwise between corresponding elements of `self`
```

File: lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/model/MethodDefinition.java [hunk 5]
```diff
@@ -198,7 +198,7 @@ public static class ArgumentDefinition {
     private static final String THUNK = "org.enso.interpreter.runtime.callable.argument.Thunk";
     private static final String CALLER_INFO = "org.enso.interpreter.runtime.callable.CallerInfo";
     private static final String DATAFLOW_ERROR = "org.enso.interpreter.runtime.error.DataflowError";
-    private static final String THIS = "this";
+    private static final String THIS = "self";
```
Comment (radeusgd) [inline lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/model/MethodDefinition.java]: Don't we want to rename `THIS` to `SELF` too?

**Review by radeusgd — APPROVED** [review]
I've skimmed through all the changes and apart from the few comments above it looks good to me.

I'd like to discuss the `that` issue, but it likely can be addressed separately from this PR, as it is already huge as it is.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso (hunk 2 above)
Comment (hubertp) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso:285]: Yes, this is something @ekmett raised as well

File: lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/model/MethodDefinition.java (hunk 5 above)
Comment (hubertp) [reply lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/model/MethodDefinition.java]: It doesn't hurt, yes, although this is Java-specific

Comment (radeusgd) [reply lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/model/MethodDefinition.java]: Ah, in that case maybe indeed it's better to keep it as-is.

**Review by JaroslavTulach — APPROVED** [review]
So many changed files! Let's get this done.

File: lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/model/MethodDefinition.java (hunk 5 above)
Comment (hubertp) [reply lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/model/MethodDefinition.java]: I think @jdunkerley kind of expected for that to happen during this PR so I will make that change

**Review by 4e6 — APPROVED** [review]
Epic!

**Comment by hubertp** [comment]
@MichaelMauderer @mwu-tow @wdanilo @farmaazon please ignore everything apart from `api/gui` stuff. No need for you to waste time on this monster PR.

**Review by wdanilo — APPROVED** [review]

**Pull request merged**